packages/cozy-doctypes/src/Folder.js

```diff
@@ -0,0 +1,75 @@
+const Application = require('./Application')
+const CozyFile = require('./File')
+
+/**
+ * Class representing the folder model.
+ * @extends CozyFile
+ */
+class CozyFolder extends CozyFile {
+  /**
+   * Returns an array of folder referenced by the given reference
+   * @param  {Object}  client    CozyClient instance
+   * @param  {String}  reference
+   * @return {Array}             Array of folders referenced with the given
+   * reference
+   */
+  static async getReferencedFolders(client, reference) {
+    const { included } = await client
+      .collection(CozyFile.doctype)
+      .findReferencedBy({
+        _type: Application.doctype,
+        _id: reference
+      })
+    return included.filter(
+      folder => !/^\/\.cozy_trash/.test(folder.attributes.path)
+    )
+  }
+
+  /**
+   * Returns an unique folder referenced with the given reference. Creates it
+   * if it does not exist.
+   * @param  {Object}  client    CozyClient instance
+   * @param  {String}  path      Path used to create folder if the referenced
+   * folder does not exist.
+   * @param  {String}  reference
```

Comment:
what is a reference ? :D

Reply:
I updated the whole code to distinguish reference to documents and our "Magic folders" logic.